ipaserver/install/dsinstance.py

```diff
@@ -1033,22 +1033,52 @@ def uninstall(self):
                 logger.error(
                     'Unable to restart DS instance %s: %s', ds_instance, e)
 
+    def get_server_cert_nickname(self, serverid=None):
+        """
+        Retrieve the nickname of the server cert used by dirsrv.
+
+        The method reads the NSS DB instead of using the attribute
+        nssslpersonalityssl of cn=RSA,cn=encryption,cn=config because
+        LDAP is not always accessible when we need to get the nickname
+        (for instance during uninstall).
```

Comment:
Is there a reason we cannot read the `nsSSLPersonalitySSL` attribute directly out of
`/etc/dirsrv/slapd-REALM-NAME/dse.ldif` ?

During uninstall, `DsInstance.stop_tracking_certificates` gets called before `remove_ds_instance`,
so `dse.ldif` should still be available to read directly.

Reply:
@frasertweedale 
I initially thought that this would be using the internal implementation of the directory server (cn=config stored in dse.ldif) but other parts of the code also directly read this file, so your idea is better.
I updated the PR with both your suggestions, thank you for the review!